File: from-1401-to-1500/1475-final-prices-with-a-special-discount-in-a-shop/solution.py
class Solution:
    def finalPrices(self, prices: List[int]) -> List[int]:
        result = prices[::]
        for i, p in enumerate(prices):
            for j in range(i+1, len(prices)):
                if prices[j] <= p:
                    result[i] = p - prices[j]
                    break
        return result

File: from-1401-to-1500/1475-final-prices-with-a-special-discount-in-a-shop/solution.go
func finalPrices(prices []int) []int {
    result := make([]int, len(prices))

    for i := 0; i < len(prices); i++ {
        price := prices[i]
        for j := i+1; j < len(prices); j++ {
            if prices[j] <= price {
                price -= prices[j]
                break
            }
        }
        result[i] = price
    }
    return result
}